### PR TITLE
OnboardingAPI: Sanitize errors

### PR DESCRIPTION
**Pull request by sophialittlejohn**
### Description

This pull request...

Closes #<issue-number>

### Approvals

- [ ] Dev
- [ ] Dev
- [ ] Designer
- [ ] Product

### CODE REVIEW

File: onboarding-api/src/controllers/emails/sendVerifyEmail.ts [hunk 1]
```diff
@@ -23,25 +23,21 @@ export const sendVerifyEmailController = async (
...
     }
 
     if (user.globalSteps.verifyEmail.completed) {
-      throw new HttpsError(400, 'Email already verified')
+      throw new HttpError(400, 'Email already verified')
     }
 
     if (email && email !== user.email) {
-      await validateAndWriteToFirestore(walletAddress, { email }, 'entity', ['email'])
+      await validateAndWriteToFirestore(walletAddress, { email }, user.investorType, ['email'])
```
Comment (sophialittlejohn) [inline onboarding-api/src/controllers/emails/sendVerifyEmail.ts:34]: email resetting works for individual users as well

File: onboarding-api/src/database/index.ts [hunk 2]
```diff
@@ -119,7 +119,7 @@ export const firestore = new Firestore()
 export const userCollection = firestore.collection(`onboarding-users`)
 
 export const storage = new Storage()
-export const onboardingBucket = storage.bucket('onboarding-api-dev')
+export const onboardingBucket = storage.bucket('onboarding-api')
```
Comment (jpangelle) [inline onboarding-api/src/database/index.ts:122]: are we ready to change this quite yet?

File: onboarding-api/src/middleware/rateLimiter.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import rateLimit from 'express-rate-limit'
+
+export const rateLimiter = rateLimit({
+  windowMs: 15 * 60 * 1000, // 15 minutes
+  max: 100, // Limit each IP to 100 requests per `window` (here, per 15 minutes)
+  standardHeaders: true, // Return rate limit info in the `RateLimit-*` headers
+  legacyHeaders: false, // Disable the `X-RateLimit-*` headers
+})
```
Comment (jpangelle) [inline onboarding-api/src/middleware/rateLimiter.ts:8]: ooooo cool! 😎

File: pinning-api/src/controllers/unpinFile.ts [hunk 4]
```diff
@@ -1,13 +1,29 @@
...
+      .replace(/[^a-z0-9]/gi, '')
+      // Escape HTML entities
+      .replace(/&/g, '&amp;')
+      .replace(/</g, '&lt;')
+      .replace(/>/g, '&gt;')
+      .replace(/"/g, '&quot;')
+      .replace(/'/g, '&#x27;')
+      .replace(/\//g, '&#x2F;')
+  )
+}
```
Comment (jpangelle) [inline pinning-api/src/controllers/unpinFile.ts:17]: Does this capture all of the cases? curious if there is a library that would be more exhaustive

File: onboarding-api/src/database/index.ts (hunk 2 above)
Comment (jpangelle) [reply onboarding-api/src/database/index.ts:122]: ah, ok once this is merge, i'll rename the bucket

File: pinning-api/src/controllers/unpinFile.ts (hunk 4 above)
Comment (sophialittlejohn) [reply pinning-api/src/controllers/unpinFile.ts]: good point, I did some research and it's definitely quite error prone to handle sanitizing manually. So I chose the library [xss](https://www.npmjs.com/package/xss) to do it for us

**Review by jpangelle — APPROVED** [review]
🛁 !!

**Pull request merged**